sktime/transformations/series/fourier.py

```diff
@@ -152,6 +165,15 @@ def _fit(self, X, y=None):
         # store the integer form of the minimum date in the prediod index
         # this is used to make sure that time t is calculated with reference to
         # the data passed on fit
+        if isinstance(X.index, pd.DatetimeIndex):
+            if X.index.freq is None:
+                if self.freq is None:
+                    ValueError("X has no known frequency and none is supplied")
```

Comment:
You can try to enforce a frequency here:
`X.index.freq = pd.infer_freq(df.index)`

We're successfully using this to overcome some mysterious pandas behaviour here:
https://github.com/sktime/sktime/blob/18b502e969d9c8d8dafce5cea78eb0c43fe23d91/sktime/datatypes/_vectorize.py#L387

you'd still need to throw the exception if `pd.infer_freq` returns None.